### PR TITLE
Trim grep expression before processing it.

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes issue that `<command>~expr > /tmp/somewhere` is differently parsed than `<command>~expr> /tmp/somewhere`

**Test plan**

All green

**Closing issues**

...

### CODE REVIEW

**Comment by Rot127** [comment]
Maybe we should abandon this and just wait until we refactor grep to use the PCRE2? Because currently it doesn't

File: librz/cons/grep.c [hunk 1]
```diff
@@ -433,11 +433,13 @@ RZ_API char *rz_cons_grep_strip(char *cmd, const char *quotestr) {
...
 
-RZ_API void rz_cons_grep_process(char *grep) {
-	if (grep) {
-		parse_grep_expression(grep);
-		free(grep);
+RZ_API void rz_cons_grep_process(RZ_OWN char *grep) {
+	if (!grep) {
+		return;
 	}
+	rz_str_trim(grep);
```
Comment (wargio) [inline librz/cons/grep.c]: i think this should be just a right trim.

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

File: librz/cons/grep.c [hunk 2]
```diff
@@ -433,11 +433,13 @@ RZ_API char *rz_cons_grep_strip(char *cmd, const char *quotestr) {
...
 
-RZ_API void rz_cons_grep_process(char *grep) {
-	if (grep) {
-		parse_grep_expression(grep);
-		free(grep);
+RZ_API void rz_cons_grep_process(RZ_OWN char *grep) {
+	if (!grep) {
+		return;
 	}
+	rz_str_trim_head(grep);
```
Comment (wargio) [inline librz/cons/grep.c]: this should be `rz_str_trim_tail`

Comment (Rot127) [reply librz/cons/grep.c]: True, agitated day.

**Comment by ret2libc** [comment]
Damn I also had the same feeling as @wargio but then I looked at the docstring and said:
```
/**
 * \brief Removes whitespace characters (space, tab, newline etc.)
 * from the end of a string.
 * The string is changed in place.
 *
 * \param str The string to trim.
 */
RZ_API void rz_str_trim_head(RZ_NONNULL char *str) {
```

Could you fix that. Of course trim_head is removing spaces from the beginning of the string, not the end.

**Pull request merged**